clients/src/main/java/org/apache/kafka/common/metrics/stats/Frequencies.java

```diff
@@ -0,0 +1,191 @@
+     *                    value is not needed
+     * @throws IllegalArgumentException if any of the {@link Frequency} objects do not have a
+     *                                  {@link Frequency#centerValue() center value} within the specified range
+     */
+    public Frequencies(int buckets, double min, double max, Frequency... frequencies) {
+        super(0.0);
+        if (min >= max) {
+            throw new IllegalArgumentException("The maximum value " + max
+                                                       + " must be greater than the minimum value " + min);
+        }
+        if (buckets < 1) {
+            throw new IllegalArgumentException("Must be at least 2 buckets");
+        }
+        if (buckets < frequencies.length) {
+            throw new IllegalArgumentException("More frequencies than buckets");
+        }
+        this.frequencies = frequencies;
+        for (Frequency freq : frequencies) {
+            if (min > freq.centerValue() || max < freq.centerValue()) {
+                throw new IllegalArgumentException("The frequency centered at '" + freq.centerValue()
+                                                           + "' is not within the range [" + min + "," + max + "]");
+            }
+        }
+        double halfBucketWidth = (max - min) / (buckets - 1) / 2.0;
+        this.binScheme = new ConstantBinScheme(buckets, min - halfBucketWidth, max + halfBucketWidth);
+    }
+
+    @Override
+    public List<NamedMeasurable> stats() {
+        List<NamedMeasurable> ms = new ArrayList<>(this.frequencies.length);
+        for (Frequency frequency : this.frequencies) {
+            final double center = frequency.centerValue();
+            ms.add(new NamedMeasurable(frequency.name(), new Measurable() {
+                public double measure(MetricConfig config, long now) {
+                    return frequency(config, now, center);
+                }
+            }));
+        }
+        return ms;
+    }
+
+    /**
+     * Return the computed frequency describing the number of occurrences of the values in the bucket for the given
+     * center point, relative to the total number of occurrences in the samples.
+     *
+     * @param config      the metric configuration
+     * @param now         the current time in milliseconds
+     * @param centerValue the value corresponding to the center point of the bucket
+     * @return the frequency of the values in the bucket relative to the total number of samples
+     */
+    public double frequency(MetricConfig config, long now, double centerValue) {
+        purgeObsoleteSamples(config, now);
+        long totalCount = 0;
+        for (Sample sample : this.samples) {
+            totalCount += sample.eventCount;
+        }
+        if (totalCount == 0) {
+            return 0.0d;
+        }
+        // Add up all of the counts in the bin corresponding to the center value
+        float count = 0.0f;
```

Comment:
I was confused about this as well, but this is the style used in other classes in the metrics framework. Notice the values in the histogram are actually `float[]`, which is why this is used here. Since we're adding multiple `float` values, it's more efficient for `count` to be `float` and to convert once at the end.